test/e2e/pkg/exec/exec_test.go

```diff
@@ -0,0 +1,84 @@
+package exec
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTSWriter(t *testing.T) {
+	// length of time added to output
+	const TimePrefixLen = 11
+	type testCase struct {
+		input []byte
+	}
+
+	testCases := []testCase{
+		{nil},
+		{[]byte{}},
+		{[]byte{'\n'}},
+		{[]byte("hi")},
+		{[]byte("hi\n")},
+		{[]byte("test\nnew\nlines\n\n\nWonder if it will work")},
+	}
+
+	for _, tc := range testCases {
+		t.Run("", func(t *testing.T) {
+			buf := bytes.Buffer{}
+
+			writer := &tsWriter{out: &buf, start: time.Now()}
+			_, err := writer.Write(tc.input)
+			assert.NoError(t, err)
+
+			out := buf.Bytes()
+			newlines := bytes.Count(tc.input, []byte{'\n'})
+			if len(tc.input) > 0 && tc.input[len(tc.input)-1] != '\n' {
+				//Add initial new line.
+				// We don't add it if last char is a new line, as it will only switch the flag to add prefix in next Write()
+				newlines++
+			}
+			assert.Len(t, out, len(tc.input)+newlines*TimePrefixLen, "new lines: %d", newlines)
```

Comment:
Don't you want to precompute the expected value along with the input?

Reply:
the benefit of this approach is obvious:
1. you always see what you want
2. easy to understand a test
(computing expected value also might be wrong, btw)

IMO: tests must have predefined expectations as much as possible. and only in exceptional cases we can use real-time computing